### ISSUE TITLE
Registering environment with static constant values

### ISSUE THREAD

**Issue by srdrymn** [Milad-Akarie/injectable#203]
I am getting 
```
type 'List<String?>' is not a subtype of type 'List<String>'
``` 
when I try to register LazySingleton with 
``` 
env: [env1, env2]
``` 
where env1 and env2 are static const String values.

Injectable version: 1.2.2

**Comment by aashir-khan** [comment]
From the error messages, seems like one or both of `env1, env2` can be null (hence the `?`) while the env parameter requires strings that are never nuull. Can you verify if those variables can be null?

**Comment by srdrymn** [comment]
This is my class where I keep my variables

```
abstract class AppEnvironment {
  static const String test = 'test';
  static const String development = 'dev';
  static const String production = 'prod';
}
```


and this is my registration line

`@LazySingleton(as: IAbstractClass, env: [AppEnvironment.test, AppEnvironment.development])`

Everything was working fine until the version 1.2.2

**Comment by aashir-khan** [comment]
The package was just yesterday updated to version 1.3.0. Can you see if that fixes the issue?

Also just a note that there is a built in `Environment` class in this package you can use to accomplish the same thing that your `AppEnvironment` class is doing

**Comment by mozochek** [comment]
This is the issue I describe in [this ](https://github.com/Milad-Akarie/injectable/issues/205)topic

**Comment by srdrymn** [comment]
> The package was just yesterday updated to version 1.3.0. Can you see if that fixes the issue?
> 
> Also just a note that there is a built in `Environment` class in this package you can use to accomplish the same thing that your `AppEnvironment` class is doing

as @mozochek described the problem still exists on 1.3.0 when I try to run build_runner.

Unfortunately I can't use `AppEnvironment` class because I need to use the environment constants that are defined in a separate domain library which does not depend on Injectable package.

**Comment by mozochek** [comment]
> > The package was just yesterday updated to version 1.3.0. Can you see if that fixes the issue?
> > Also just a note that there is a built in `Environment` class in this package you can use to accomplish the same thing that your `AppEnvironment` class is doing
> 
> as @mozochek described the problem still exists on 1.3.0 when I try to run build_runner.
> 
> Unfortunately I can't use `AppEnvironment` class because I need to use the environment constants that are defined in a separate domain library which does not depend on Injectable package.

You can fix problem with 2 easy ways: rool back versions or edit local copy of package files

**Comment by Milad-Akarie** [comment]
fixed in version 1.4.1
sorry for the delay.
